test/CMakeLists.txt

```diff
@@ -17,7 +17,8 @@ ExternalProject_Add(fakeit
         GIT_TAG master
         UPDATE_COMMAND ${GIT_EXECUTABLE} pull
         CONFIGURE_COMMAND ""
-        BUILD_COMMAND ""
+	PATCH_COMMAND cp ${gram_SOURCE_DIR}/patch/fakeit.hpp ${gram_SOURCE_DIR}/test/lib/fakeit/src/fakeit/single_header/catch/
```

Comment:
What's the reason you switched from `patch` to `cp` in your last commit? I'd prefer to have a small patch instead of the whole header file in the Gram repo.

Reply:
I switched from patch, because this version worked only localy. Patch supplied in that commit won't work, becuase of timestamp created by diff. Also couldn't use `git diff` and `git apply`, because fakeit repository is not part of gram repository and thus git prevents such changes.

Reply:
I don't think the patch needs a timestamp. If you copy the appropriate lines from the [PR patch file](https://patch-diff.githubusercontent.com/raw/eranpeer/FakeIt/pull/201.patch), the patch should apply cleanly.

Also, my version of CMake (3.18.4) runs `PATCH_COMMAND` within the FakeIt repository, so something like `PATCH_COMMAND ${GIT_EXECUTABLE} apply ${gram_SOURCE_DIR}/patch/fakeit.hpp.patch` appears to work fine. Are you seeing something different?

The problem then is that CMake runs the patch command on each build. Because Git cannot apply the patch more than once, it exits with an error code on subsequent builds and causes a failure. I think this can be solved by discarding the return value by appending `|| true` to the patch command.

Reply:
This aproach somehow previously failed, but now when I did a clean build it is resolved and works just fine. Also agree on using `|| true` at the end of `PATCH_COMMAND`to ensure applying patch multiple times doesn't cause build failure.

I also noticed catch `GIT_TAG` needs to be reverted back to just `v2.x`, because `origin/v2.x` causes detached HEAD state and thus forbids `git pull`.

Reply:
> This aproach somehow previously failed, but now when I did a clean build it is resolved and works just fine. Also agree on using `|| true` at the end of `PATCH_COMMAND`to ensure applying patch multiple times doesn't cause build failure.

Great.

> I also noticed catch `GIT_TAG` needs to be reverted back to just `v2.x`, because `origin/v2.x` causes detached HEAD state and thus forbids `git pull`.

You might need to rebase, I removed `git pull` in c9046fdfec674865adc0c736998193548c80b28d. In any case, I would prefer to see a specific tag rather than a branch for Catch2. It seems like the safer option given that FakeIt depends on its internals that could change at any time and break Gram again.